### PR TITLE
add new sns topic and slack integration

### PR DESCRIPTION
**Pull request by dorianjp**
https://trello.com/c/4e58AlOx/178-sns-warning-topic-should-only-be-used-by-alarms

* Added new topic "notification-canada-ca-alert-general"
* Re-assigned to new topic:
  * aws_health
  * rds
* Added new slack module "notify_slack_general"
  * bind to new general sns topic
  * new env vars for webhook and channel, copied values similar to existing slack integrations

### CODE REVIEW

**Comment by dorianjp** [comment]
[SNS warning Topic should only be used by Alarms](https://trello.com/c/4e58AlOx/178-sns-warning-topic-should-only-be-used-by-alarms)

**Comment by dorianjp** [comment]
@AntoineAugusti I left email out as the email subscription protocol is [unsupported](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/sns_topic_subscription#email) unless you have something else in mind to send the email, or just do it this way regardless. I thought about using notify via https, but that wouldn't work with some of the failure scenarios, after-all notify would be potentially down 😄

File: .github/workflows/merge_to_main_production.yml [hunk 1]
```diff
@@ -27,8 +27,10 @@ env:
   TF_VAR_rds_cluster_password: ${{ secrets.PRODUCTION_RDS_CLUSTER_PASSWORD }}
   TF_VAR_cloudwatch_slack_webhook_warning_topic: ${{ secrets.PRODUCTION_CLOUDWATCH_SLACK_WEBHOOK }}
   TF_VAR_cloudwatch_slack_webhook_critical_topic: ${{ secrets.PRODUCTION_CLOUDWATCH_SLACK_WEBHOOK }}
+  TF_VAR_cloudwatch_slack_webhook_general_topic: ${{ secrets.PRODUCTION_CLOUDWATCH_SLACK_WEBHOOK }}
   TF_VAR_slack_channel_warning_topic: "notification-ops"
   TF_VAR_slack_channel_critical_topic: "notification-ops"
+  TF_VAR_slack_channel_general_topic: "notification-ops"
```
Comment (AntoineAugusti) [inline .github/workflows/merge_to_main_production.yml:33]: These changes are fine, but you need to submit them later. If you submit them now, the production CI job will try to assign these variables in the codebase @ version 0.54.0 as defined in `INFRASTRUCTURE_VERSION` a few lines before and these variables don't exist yet.

You'll need to submit these changes when bumping the production version.

File: aws/rds/variables.tf [hunk 2]
```diff
@@ -19,6 +19,6 @@ variable "vpc_private_subnets" {
   type = list(any)
 }
 
-variable "sns_alert_warning_arn" {
+variable "sns_alert_general_arn" {
```
Comment (AntoineAugusti) [inline aws/rds/variables.tf:22]: You'll need to change the file `env/staging/rds/terraform.hcl` to map this variable to the appropriate value in the `common` module

File: aws/eks/variables.tf [hunk 3]
```diff
@@ -42,6 +42,10 @@ variable "sns_alert_critical_arn" {
   type = string
 }
 
+variable "sns_alert_general_arn" {
```
Comment (AntoineAugusti) [inline aws/eks/variables.tf:45]: You'll need to change the file `env/staging/eks/terraform.hcl` to map this variable to the appropriate value in the `common` module

**Review by AntoineAugusti — CHANGES_REQUESTED** [review]

**Comment by AntoineAugusti** [comment]
@dorianjp Yeah, the only way to send alerts via email at the moment would be to have an SNS subscription to a Lambda and then you write a Lambda to send an email with SES or Notify so you can leave that out for now

File: .github/workflows/merge_to_main_production.yml (hunk 1 above)
Comment (dorianjp) [reply .github/workflows/merge_to_main_production.yml]: Sounds good, took the prod vars out for now.

File: aws/common/slack.tf [hunk 4]
```diff
@@ -34,3 +34,21 @@ module "notify_slack_critical" {
...
+
+  slack_webhook_url = var.cloudwatch_slack_webhook_general_topic
+  slack_channel     = var.slack_channel_general_topic
+  slack_username    = "[GENERAL] AWS Cloudwatch"
+  slack_emoji       = ":loudspeaker:"
+
+  lambda_function_name                   = "notify-slack-general"
+  cloudwatch_log_group_retention_in_days = 90
+
+  depends_on = [aws_sns_topic.notification-canada-ca-alert-general]
```
[the remark this answers is not in the document]
Comment (dorianjp) [reply aws/common/slack.tf]: Done

File: aws/eks/variables.tf (hunk 3 above)
Comment (dorianjp) [reply aws/eks/variables.tf:45]: Assuming terragrunt.hcl?
Done

File: aws/rds/variables.tf (hunk 2 above)
Comment (dorianjp) [reply aws/rds/variables.tf:22]: +1

**Comment by dorianjp** [comment]
@AntoineAugusti , I have a few changes I will need to push on the next 2 PRs due to the CI failing on expected resources:
* TF_VARs discussed above (during prod push only)
* slack.tf -> the definition of the general alerting module (aws plan looks for the sns topic prematurely when this is in)
* terragrunt.hcl (eks and rds) adding the input bindings

**Review by AntoineAugusti — APPROVED** [review]
LGTM.

Please make sure that the commit message will be prefixed with `feat:` when merging your PR so that a new tag will be created.

Looking forward to the other PRs 👍

**Pull request merged**